rqt_bag/src/rqt_bag/bag_widget.py

```diff
@@ -289,36 +296,43 @@ def load_bag(self, filename):
         # self.progress_bar.setTextVisible(True)
 
         try:
-            bag = rosbag.Bag(filename)
-            self.play_button.setEnabled(True)
-            self.thumbs_button.setEnabled(True)
-            self.zoom_in_button.setEnabled(True)
-            self.zoom_out_button.setEnabled(True)
-            self.zoom_all_button.setEnabled(True)
-            self.next_button.setEnabled(True)
-            self.previous_button.setEnabled(True)
-            self.faster_button.setEnabled(True)
-            self.slower_button.setEnabled(True)
-            self.begin_button.setEnabled(True)
-            self.end_button.setEnabled(True)
-            self.save_button.setEnabled(True)
-            self.record_button.setEnabled(False)
-            self._timeline.add_bag(bag)
-            qDebug("Done loading '%s'" % filename.encode(errors='replace'))
-            # put the progress bar back the way it was
-            self.set_status_text.emit("")
-        except rosbag.ROSBagException as e:
+            with open(filename) as f:
+                bag_info = yaml.load(f)
+                bag = Rosbag2(bag_info['rosbag2_bagfile_information'], filename)
+        except Exception as e:
             qWarning("Loading '%s' failed due to: %s" % (filename.encode(errors='replace'), e))
             self.set_status_text.emit("Loading '%s' failed due to: %s" % (filename, e))
+            return
+
+        qDebug('Loading bag from metadata file "{}" Succeeded'.format(filename))
+
+        self.play_button.setEnabled(True)
+        self.thumbs_button.setEnabled(True)
+        self.zoom_in_button.setEnabled(True)
+        self.zoom_out_button.setEnabled(True)
+        self.zoom_all_button.setEnabled(True)
+        self.next_button.setEnabled(True)
+        self.previous_button.setEnabled(True)
+        self.faster_button.setEnabled(True)
+        self.slower_button.setEnabled(True)
+        self.begin_button.setEnabled(True)
+        self.end_button.setEnabled(True)
+        self.save_button.setEnabled(True)
+        self.record_button.setEnabled(False)
+        self._timeline.add_bag(bag)
+        qDebug("Done loading '%s'" % filename.encode(errors='replace'))
+        # put the progress bar back the way it was
+        self.set_status_text.emit("")
 
         # self.progress_bar.setFormat(progress_format)
         # self.progress_bar.setTextVisible(progress_text_visible) # causes a segfault :(
         # self.progress_bar.setRange(0, 100)
         # self clear loading filename
 
     def _handle_save_clicked(self):
-        filename = QFileDialog.getSaveFileName(
-            self, self.tr('Save selected region to file...'), '.', self.tr('Bag files {.bag} (*.bag)'))
+        filename = \
+            QFileDialog.getSaveFileName(self, self.tr('Save selected region to file...'), '.',
+                                        self.tr('Bag files {.bag} (*.bag)'))
```

Comment:
tiny style nit: breaking after the first open parenthesis is more inline with our [style guide (prefer hanging indents](https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#id4).
```suggestion
        filename = QFileDialog.getSaveFileName(
            self,
            self.tr('Save selected region to file...'),
            '.',
            self.tr('Bag files {.bag} (*.bag)'))
```

Breaking each argument onto it's own line is optional. Feel free to group two or more onto the same line.